packages/core/src/utils/index.ts

```diff
@@ -68,15 +68,28 @@ builder.objectType('MyObject', {
   }
 }
 
-export function brandWithType<Types extends SchemaTypes>(val: unknown, type: OutputType<Types>) {
+type BrandWithTypeResult<T, Types extends SchemaTypes> = T extends Record<string, unknown>
+  ? {
+      [K in keyof T & typeof typeBrandKey]: K extends typeof typeBrandKey
```

Comment:
This doesn't look like it would do what you expect.  Joining with an intersection here probably produces never

Reply:
Right, just realized that 😂 Maybe it should be an union...